.github/workflows/commit-checker.yaml

```diff
@@ -0,0 +1,46 @@
+name: Unit Tests
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - closed
+      - synchronize
+  workflow_dispatch:
+env:
+  RESOURCES_DIR: ${{ github.workspace }}/.github/resources
+  GH_USER_EMAIL: 140449482+dsp-developers@users.noreply.github.com
+  GH_USER_NAME: dsp-developers
+jobs:
+    commit_checker:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                fetch-depth: 0
+            - name: Get Commits
+              id: get-commits
+              run: |
+                master_commit=$(cat .git/refs/remotes/origin/master)
+                echo "master_commit_hash=$master_commit" >> $GITHUB_OUTPUT
+                last_commit=$(cat .git/refs/remotes/pull/${{ github.event.pull_request.number }}/merge)
+                echo "last_commit_hash=$last_commit" >> $GITHUB_OUTPUT
+            - name: Run Commit Checker
+              shell: bash
+              env:
+                GH_TOKEN: ${{ secrets.GH_TOKEN_PROJECT_EDIT }}
```

Comment:
From your CI output, it looks like the secret isn't getting stored in this env var.
 pull_request trigger event. Looks like secrets can be accessed when workflow is triggered by this event:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

> secrets are not passed to the runner when a workflow is triggered from a forked repository. The GITHUB_TOKEN has read-only permissions in pull requests from forked repositories.

You might want to try this event: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

> For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets